src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishBuildToMaestro.cs

```diff
@@ -130,7 +132,9 @@ public async Task<bool> PushMetadataAsync(CancellationToken cancellationToken)
                             return !Log.HasLoggedErrors;
                         }
 
-                        buildModel = _buildModelFactory.CreateMergedModel(parsedManifests, ArtifactVisibility.All);
+                        // Always drop Internal artifacts when pushing a build to Maestro.
+                        buildModel = _buildModelFactory.CreateMergedModel(parsedManifests,
+                            (ArtifactVisibility.All & ~ArtifactVisibility.Internal));
```

Comment:
Any reason to not change this to just publish External assets?

```suggestion
                            ArtifactVisibility.External);
```